### PR TITLE
Add library to interact with operator conditions

### PR DESCRIPTION
**Pull request by varshaprasad96**
**Description of the change:**
This PR removes the previous operator_conditions package and
introduces a `Conditions` interface, using which one can:
1. Get an existing condition from the operator conditions CR in cluster.
2. Create/Update a condition in the operator condition CR.

**Motivation for the change:**
EP: [Operator-Conditions lib](https://github.com/operator-framework/enhancements/blob/master/enhancements/operator-conditions-lib.md)

### CODE REVIEW

**Review by jmrodri — APPROVED** [review]
/lgtm
This looks a lot better. Great work!

File: testbin/setup-envtest.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
```
Comment (estroz) [inline testbin/setup-envtest.sh:1]: Nit: this file should be ignored by adding `**/testbin/*` to the `.gitignore`.

**Pull request merged**